plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2648,6 +2648,11 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
                     if (!(handle instanceof IcebergTableHandle icebergHandle)) {
                         return UNKNOWN_SNAPSHOT_TOKEN;
                     }
+                    // Currently the catalogs are isolated in separate classloaders, and the above instanceof check is sufficient to know "our" handles.
+                    // This isolation will be removed after we remove Hadoop dependencies, so check that this is "our" handle explicitly.
+                    if (!trinoCatalogHandle.equals(icebergHandle.getCatalog())) {
+                        return UNKNOWN_SNAPSHOT_TOKEN;
```

Comment:
It might be worth documenting for users that source iceberg tables belonging to a different iceberg catalog will be treated like non-iceberg tables. A user might expect that we will take advantage of snapshot id in refresh for all iceberg catalogs.

Reply:
good idea! 
cc @bitsondatadev @colebow  do you want to help here?